app/src/androidTest/java/com/android/voyageur/ui/overview/AddTripTest.kt

```diff
@@ -0,0 +1,93 @@
+    tripsViewModel = TripsViewModel(tripRepository)
+
+    `when`(navigationActions.currentRoute()).thenReturn(Screen.ADD_TRIP)
+    whenever(tripsViewModel.getNewTripId()).thenReturn("mockTripId")
+  }
+
+  private fun <T> anyNonNull(): T = Mockito.any<T>()
+
+  @Test
+  fun addTripScreen_initialState() {
+    composeTestRule.setContent { AddTripScreen(tripsViewModel, navigationActions) }
+
+    composeTestRule.onNodeWithTag("inputTripTitle").assertExists()
+    composeTestRule.onNodeWithTag("inputTripDescription").assertExists()
+    composeTestRule.onNodeWithTag("inputTripCreator").assertExists()
+    composeTestRule.onNodeWithTag("inputTripParticipants").assertExists()
+    composeTestRule.onNodeWithTag("inputTripLocations").assertExists()
+    composeTestRule.onNodeWithTag("inputStartDate").assertExists()
+    composeTestRule.onNodeWithTag("inputEndDate").assertExists()
+    composeTestRule.onNodeWithTag("tripSave").assertExists()
+
+    composeTestRule.onNodeWithTag("addTripTitle").assertTextEquals("Create a New Trip")
+    composeTestRule.onNodeWithTag("tripSave").assertTextEquals("Save Trip")
+  }
+
+  @Test
+  fun addTripScreen_inputsUpdateState() {
+    composeTestRule.setContent { AddTripScreen(tripsViewModel, navigationActions) }
+
+    composeTestRule.onNodeWithTag("inputTripTitle").performTextInput("London")
+    composeTestRule.onNodeWithTag("inputTripDescription").performTextInput("4 days in London")
+    composeTestRule.onNodeWithTag("inputTripCreator").performTextInput("John Doe")
+    composeTestRule.onNodeWithTag("inputTripParticipants").performTextInput("Alice, Bob")
+
+    composeTestRule.onNodeWithTag("inputTripTitle").assertExists()
+    composeTestRule.onNodeWithTag("inputTripDescription").assertExists()
+    composeTestRule.onNodeWithTag("inputTripCreator").assertExists()
+    composeTestRule.onNodeWithTag("inputTripParticipants").assertExists()
+  }
+
+  @Test
+  fun doesNotSubmitWithInvalidDate() {
+    composeTestRule.setContent { AddTripScreen(tripsViewModel, navigationActions) }
+
+    composeTestRule.onNodeWithTag("inputStartDate").performTextClearance()
+    composeTestRule.onNodeWithTag("inputStartDate").performTextInput("notadate")
+
+    composeTestRule.onNodeWithTag("inputEndDate").performTextClearance()
+    composeTestRule.onNodeWithTag("inputEndDate").performTextInput("05/01/2024")
+
+    composeTestRule.onNodeWithTag("tripSave").performClick()
+
+    verify(tripRepository, never()).updateTrip(anyNonNull(), anyNonNull(), anyNonNull())
+  }
+
+  @Test
+  fun addTripScreen_tripTypeSelection() {
+    composeTestRule.setContent { AddTripScreen(tripsViewModel, navigationActions) }
+
+    composeTestRule.onNodeWithTag("tripTypeBusiness").performClick()
+    assert(tripsViewModel.tripType.value == TripType.BUSINESS)
+
+    composeTestRule.onNodeWithTag("tripTypeTourism").performClick()
+    assert(tripsViewModel.tripType.value == TripType.TOURISM)
+  }
+}
```

Comment:
I checked out the code coverage report and noticed the timestamp is directly converted in the AddTripScreen. I would suggest making a separate method to convert the timestamp, and then writing a test for that method. Or at least writing a test which checks that the timestamp is correctly converted.